scripts/tool_benchmarks.py

```diff
@@ -0,0 +1,188 @@
+import datetime
+import uuid
+from typing import Callable, List, cast
+
+from langchain.chat_models import init_chat_model
+from langchain_chroma import Chroma
+from langchain_core.documents import Document
+from langchain_core.messages import HumanMessage, SystemMessage
+from langchain_core.messages.utils import convert_to_messages
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_core.tools import BaseTool, tool
+from langchain_openai import OpenAIEmbeddings
+from langsmith.client import Client
+
+from langchain_benchmarks import __version__
+from langchain_benchmarks.rate_limiting import RateLimiter
+from langchain_benchmarks.tool_usage.agents import StandardAgentFactory
+from langchain_benchmarks.tool_usage.tasks.multiverse_math import (
+    MULTIVERSE_MATH,
+    add,
+    cos,
+    divide,
+    log,
+    multiply,
+    negate,
+    pi,
+    power,
+    sin,
+    subtract,
+)
+
+client = Client()
+experiment_uuid = uuid.uuid4().hex[:4]
+today = datetime.date.today().isoformat()
+task = MULTIVERSE_MATH
+
+funcs: List[Callable] = [
+    multiply,
+    add,
+    divide,
+    subtract,
+    power,
+    log,
+    negate,
+    sin,
+    cos,
+    pi,
+]
+tools: List[BaseTool] = [cast(BaseTool, tool(f)) for f in funcs]
+
+models_to_test = [
+    "claude-3-haiku-20240307",
+    "claude-3-sonnet-20240229",
+    "gpt-3.5-turbo-0125",
+    "gpt-4-turbo-2024-04-09",
+    "accounts/fireworks/models/firefunction-v2",
+]
+
+examples = list(
+    client.list_examples(dataset_name="multiverse-math-examples-for-few-shot")
+)
+few_shot_messages = []
+questions = []
+for i, ex in enumerate(examples):
+    converted_messages = convert_to_messages(ex.outputs["output"])
+    questions.append(Document(converted_messages[1].content, metadata={"index": i}))
+    few_shot_messages += converted_messages
+
+few_shot_messages = [m for m in few_shot_messages if not isinstance(m, SystemMessage)]
+
+few_shot_str = ""
+for m in few_shot_messages:
+    if isinstance(m.content, list):
+        few_shot_str += "AI message: "
+        for tool_use in m.content:
+            if "name" in tool_use:
+                few_shot_str += f"Use tool {tool_use['name']}, input: {', '.join(f'{k}:{v}' for k,v in tool_use['input'].items())}"
+            else:
+                few_shot_str += tool_use["text"]
+            few_shot_str += "\n"
+    else:
+        if isinstance(m, HumanMessage):
+            few_shot_str += f"Human message: {m.content}"
+        else:
+            few_shot_str += f"AI message: {m.content}"
+
+    few_shot_str += "\n"
+
+vectorstore = Chroma.from_documents(
+    questions, embedding=OpenAIEmbeddings(model="text-embedding-3-small")
+)
+retriever = vectorstore.as_retriever()
+
+prompts = [
```

Comment:
could we upload all these prompts to the prompt hub and pull from it in the script?